packages/block-library/src/template-part/edit/index.js

```diff
@@ -64,56 +125,19 @@ export default function TemplatePartEdit( {
 		}
 	}, [ innerBlocks ] );
 
-	const blockWrapperProps = useBlockWrapperProps();
-
 	if ( postId ) {
 		// Part of a template file, post ID already resolved.
 		return (
-			<TagName { ...blockWrapperProps }>
-				<BlockControls>
-					<div className="wp-block-template-part__block-control-group">
-						<TemplatePartNamePanel
-							postId={ postId }
-							setAttributes={ setAttributes }
-						/>
-						<Dropdown
-							className="wp-block-template-part__preview-dropdown-button"
-							contentClassName="wp-block-template-part__preview-dropdown-content"
-							position="bottom right left"
-							renderToggle={ ( { isOpen, onToggle } ) => (
-								<ToolbarButton
-									aria-expanded={ isOpen }
-									icon={ isOpen ? chevronUp : chevronDown }
-									label={ __( 'Choose another' ) }
-									onClick={ onToggle }
-									// Disable when open to prevent odd FireFox bug causing reopening.
-									// As noted in https://github.com/WordPress/gutenberg/pull/24990#issuecomment-689094119 .
-									disabled={ isOpen }
-								/>
-							) }
-							renderContent={ ( { onClose } ) => (
-								<TemplatePartSelection
-									setAttributes={ setAttributes }
-									onClose={ onClose }
-								/>
-							) }
-						/>
-					</div>
-				</BlockControls>
-				<TemplatePartInnerBlocks
-					postId={ postId }
-					hasInnerBlocks={ innerBlocks.length > 0 }
-				/>
-			</TagName>
+			<Content
```

Comment:
Maybe it's just me but I don't see what these two components bring aside indirection.

Reply:
The problem is that you cannot use `useBlockWrapperProps` without having an element that set a ref that you can add event listeners to.

Reply:
I guess it's the "return null" that causes the issue.

Reply:
Right. If you have multiple paths that are all creating a block, it would be fine to "share" the hook. Not sure if in this care we can just return a placeholder block or something. A block not returning anything is strange. There's a block, but you can't see it?

Reply:
yes, it's probably some hidden bug somewhere there but can be handled separately tbh.

Reply:
@youknowriad I reverted the previous change and now it will always render a block. If the post ID is not resolved it will output "Loading..." Not sure if that's the best, but we can change the loading thing later on if needed. Imo we should always render a block even when it's still loading.

Reply:
You're right, good fix, thanks for the update.

Reply:
Ugh, some tests are failing now. :)